### PR TITLE
Radio command improvement

### PR DESCRIPTION
**Pull request by thetazero**
Asyncio commands later

### CODE REVIEW

**Review by jbwillis — CHANGES_REQUESTED** [review]
After line 128 of `Tasks/radio.py`, should there be `self.cmsg_last = response`?

**Comment by thetazero** [comment]
> After line 128 of `Tasks/radio.py`, should there be `self.cmsg_last = response`?

Yes, fixed these things

File: state_machine/applications/flight/lib/radio_utils/memory_buffered_message.py [hunk 1]
```diff
@@ -11,7 +12,7 @@ class NaiveMessage(Message):
     :type str: str | bytes | bytearray
     """
 
-    packet_len = 249  # not 251 because for some reason packet loss is extremely high at this threshold
+    packet_len = PACKET_DATA_LEN
```
Comment (jbwillis) [inline state_machine/applications/flight/lib/radio_utils/memory_buffered_message.py:15]: I noticed that this is 251 in the __init__.py. Should it be 249?

Comment (thetazero) [reply state_machine/applications/flight/lib/radio_utils/memory_buffered_message.py:15]: Max packet length is 256, lora header is 4 bytes, our extra header is 1 byte.
However I found that LORA doesn't seem to work with such large packet sizes, when doing field testing I often reduce the "MAX_PACKET_LENGTH" from 252 to around 200.

I think this is the correct value, we will probably lower MAX_PACKET_LEN however (if we can even get LORA to work)

File: state_machine/applications/flight/Tasks/radio.py [hunk 2]
```diff
@@ -64,29 +66,29 @@ async def main_task(self):
                 self.debug(f'Recieved msg "{response}", RSSI: {cubesat.radio.last_rssi - 137}')
 
                 if header == headers.NAIVE_START or header == headers.NAIVE_MID or header == headers.NAIVE_END:
```
Comment (jbwillis) [inline state_machine/applications/flight/Tasks/radio.py:68]: These should be changed to something like `MEMORY_BUFFERED_START`, `_MID` and `_END`

File: state_machine/applications/flight/Tasks/radio.py [hunk 3]
```diff
@@ -116,17 +118,17 @@ def handle_command(self, response):
             self.debug('invalid command!')
             cubesat.radio.send(b'invalid cmd' + cmd)
 
-    def handle_chunk(self, header, response):
-        """Handler function for the chunk message type"""
+    def handle_disk_buffered_message(self, header, response):
+        """Handler function for the disk_buffered_message message type"""
         if header == headers.CHUNK_START:
```
Comment (jbwillis) [inline state_machine/applications/flight/Tasks/radio.py:123]: `DISK_BUFFERED_START`

File: state_machine/applications/flight/lib/radio_utils/commands.py [hunk 4]
```diff
@@ -133,15 +134,32 @@ def delete_file(task, file):
...
+
+async def reload(task):
+    """Reloads the flight software
+
+    :param task: The task that called this function
+    """
+    task.debug('Reloading')
+    msg = bytearray([headers.DEFAULT])
+    msg.append(b'reset')
+    await cubesat.radio.send(data=msg)
```
Comment (jbwillis) [inline state_machine/applications/flight/lib/radio_utils/commands.py:146]: Doesn't seem like we'd want the software to switch to a different thread here. Is there a way to not `await` this, but force the software to wait until it completes?

File: state_machine/applications/flight/lib/radio_utils/commands.py [hunk 5]
```diff
@@ -22,18 +23,19 @@
...
 
-async def hreset(self):
+def hreset(self):
     """Hard reset"""
     self.debug('Resetting')
-    msg = bytearray([headers.DEFAULT])
-    msg.append(b'reset')
-    await cubesat.radio.send(data=msg)
-    cubesat.micro.on_next_reset(self.cubesat.micro.RunMode.NORMAL)
+    # msg = bytearray([headers.DEFAULT])
```
Comment (jbwillis) [inline state_machine/applications/flight/lib/radio_utils/commands.py:35]: delete commented out code

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/applications/flight/lib/radio_utils/commands.py (hunk 4 above)
Comment (thetazero) [reply state_machine/applications/flight/lib/radio_utils/commands.py:146]: Yes, will require a slight rewrite of the driver, will address in separate pull.

**Review by jbwillis — APPROVED** [review]

**Pull request merged**